home/gettingstarted/0.7/javausage.md

```diff
@@ -82,40 +133,58 @@ Finally, we declare the code snippet for the test by implementing the `snippet`
     }
 
 The complete test is here:
-
-    public class MyHistogramTest extends OfflineReport {
-      public Persistor javaPersistor() {
-        return new SerializationPersistor();
-      }
-      public class List implements Group {
-        public class groupBy
-        implements Using<LinkedList<Integer>, HashMap<Integer, LinkedList<Integer>>> {
-          public final JContext config = JContext.create()
-            .put("exec.benchRuns", 20)
-            .put("exec.independentSamples", 1)
-            .put("exec.outliers.covMultiplier", 1.5)
-            .put("exec.outliers.suspectPercent", 40);
-          public JavaGenerator<LinkedList<Integer>> generator() {
-            JavaGenerator<Integer> sizes = new SingleGen("size", 5000000);
-            return new CollectionGenerators(sizes).lists();
-          }
-          public HashMap<Integer, LinkedList<Integer>> 
-            snippet(LinkedList<Integer> in) {
-            HashMap<Integer, LinkedList<Integer>> hm =
-              new HashMap<Integer, LinkedList<Integer>>();
-            for (int i = 0; i < 10; i++) {
-              hm.put(i, new LinkedList<Integer>());
-            }
-            Iterator<Integer> it = in.iterator();
-            while (it.hasNext()) {
-              Integer element = it.next();
-              LinkedList<Integer> tmp = hm.get(element % 10);
-              tmp.add(element);
-              hm.put(element % 10, tmp);
+  
+    @scopes({
+      @scopeCtx(scope = "list", context = "config")
+    })
+    public class MyHistogramTest extends JBench.OfflineReport {
+      public final Context config = new ContextBuilder()
+          .put("exec.benchRuns", 20)
+          .put("exec.independentSamples", 1)
+          .put("exec.outliers.covMultiplier", 1.5)
+          .put("exec.outliers.suspectPercent", 40)
+          .build();
+    
+      public final JGen<Integer> sizes = JGen.intValues("size", 500000);
+    
+      public final JGen<Tuple2<Integer, LinkedList<Integer>>> lists = sizes.zip(sizes.map(
```

Comment:
We might need to wrap this line, because it might escape the <code> element in the generated HTML.